src/reanimated2/animation/util.ts

```diff
@@ -43,6 +44,29 @@ if (__DEV__ && IS_REDUCED_MOTION) {
   );
 }
 
+export function assertEasingIsWorklet(
+  easing: EasingFunction | EasingFunctionFactory
+) {
+  'worklet';
+
+  if (isWeb()) {
+    /* It is possible to run reanimated on web without plugin, so let's skip this check on web */
+    return true;
+  }
+  const isFunction: boolean = typeof easing === 'function';
+  const functionName = typeof easing === 'function' ? easing?.name : '';
+  /* Worklets ran on UI thread are bound first. Therefore if a function wasn't bound it cannot be a worklet. See `valueUnpacker` code for reference. */
```

Comment:
I feel bad for reworking this comment over and over, but I want to make it as informative as possible.

```suggestion
  /* Just checking with `isWorkletFunction` is not sufficient in case of Keyframe animations. Therefore, we also check for binding. */
```